cdc/log.cc

```diff
@@ -310,110 +310,211 @@ db_context db_context::builder::build() {
     };
 }
 
-/* Find some timestamp inside the given mutation.
- *
- * If this mutation was created using a single insert/update/delete statement, then it will have a single,
- * well-defined timestamp (even if this timestamp occurs multiple times, e.g. in a cell and row_marker).
+/* Given a timestamp, generates a timeuuid with the following properties:
+ * 1. `t1` < `t2` implies timeuuid_type->less(timeuuid_type->decompose(generate_timeuuid(`t1`)),
+ *                                            timeuuid_type->decompose(generate_timeuuid(`t2`))),
+ * 2. utils::UUID_gen::micros_timestamp(generate_timeuuid(`t`)) == `t`.
  *
- * This function shouldn't be used for mutations that have multiple different timestamps: the function
- * would only find one of them. When dealing with such mutations, the caller should first split the mutation
- * into multiple ones, each with a single timestamp.
+ * If `t1` == `t2`, then generate_timeuuid(`t1`) != generate_timeuuid(`t2`),
+ * with unspecified nondeterministic ordering.
  */
-// TODO: We need to
-// - in the code that calls `augument_mutation_call`, or inside `augument_mutation_call`,
-//   split each mutation to a set of mutations, each with a single timestamp.
-// - optionally: here, throw error if multiple timestamps are encountered (may degrade performance).
 // external linkage for testing
-api::timestamp_type find_timestamp(const schema& s, const mutation& m) {
-    auto& p = m.partition();
-    api::timestamp_type t = api::missing_timestamp;
+utils::UUID generate_timeuuid(api::timestamp_type t) {
+    return utils::UUID_gen::get_random_time_UUID_from_micros(t);
+}
 
-    t = p.partition_tombstone().timestamp;
-    if (t != api::missing_timestamp) {
-        return t;
-    }
+struct atomic_column_update {
+    column_id id;
+    atomic_cell cell;
```

Comment:
Can't this be an atomic_cell_view? We are keeping the original mutation alive and non-moved during the operations, no?